subprojects/core/src/integTest/groovy/org/gradle/api/file/FileCollectionSymlinkIntegrationTest.groovy

```diff
@@ -65,6 +66,59 @@ class FileCollectionSymlinkIntegrationTest extends AbstractIntegrationSpec {
         "project.objects.fileCollection()"   | "project.objects.fileCollection().from(file, symlink, symlinked)"
     }
 
+    // TODO should probably be moved to incremental suite
+    def "task with broken input #fileType symlink doesn't fail"() {
+        def target = file('target')
+        def link = file('link').createLink(target)
+        def output = file('output')
+
+        buildFile << """
+            class SymlinkedTask extends DefaultTask {
+                @Incremental ${inputAnnotation} File linkInput
+                @OutputFile File output
+            
+                @TaskAction execute(InputChanges inputChanges) {
+                    output.text = inputChanges.allFileChanges.toList()
+                }
+            }   
+            
+            task resolveSymlink(type: SymlinkedTask) {
+                linkInput = file("${escapeString(link)}")
+                output = file("${escapeString(output)}")
+            }
+        """
+
+        when:
+        run 'resolveSymlink'
+        then:
+        noneSkipped()
+        outputContains 'broken symbolic link'
+        output.text.contains "${link.name} added for rebuild"
+        when:
+        run 'resolveSymlink'
+        then:
+        allSkipped()
+
+        when:
+        create(target)
+        run 'resolveSymlink'
+        then:
+        noneSkipped()
+        output.text.contains "${link.name} has been added"
+
+        when:
+        delete(target)
+        run 'resolveSymlink'
+        then:
+        noneSkipped()
+        output.text.contains "${link.name} has been removed"
+
+        where:
+        fileType    | inputAnnotation   | create              | delete
+        'file'      | '@InputFile'      | { it.createFile() } | { it.delete() }
+        // TODO 'directory' | '@InputDirectory' | { it.mkdirs() }     | { it.deleteDir() }
```

Comment:
is this supposed to work with `InputChanges`?

Reply:
It should. Can you elaborate on what doesn't work?

Reply:
The `file` produces an initial change for `allFileChanges` of type `NonIncrementalInputChanges$RebuildFileChange`, the `folder` produces an empty iterable.